core/internal/features_test.go

```diff
@@ -902,7 +922,86 @@ func TestIntegration_FluxMonitor_Deviation(t *testing.T) {
+		eth.Register("eth_getTransactionReceipt", confirmedReceipt)
+		eth.Register("eth_getBalance", "0x100")
+		eth.Register("eth_call", "0x100")
+	})
+	newHeads <- ethpkg.BlockHeader{Number: cltest.BigHexInt(10)}
+
+	// 6. Check the FM price on completed run output
 	jr := cltest.WaitForJobRunToComplete(t, app.GetStore(), jrs[0])
-	assert.Equal(t, "102", cltest.MustResultString(t, jr.Result))
+
+	overrides := jr.Overrides
+	assert.Equal(t, "102", overrides.Get("result").String())
+	assert.Equal(
+		t,
+		"0x3cCad4715152693fE3BC4460591e3D3Fbd071b42", // from testdata/flux_monitor_job.json
+		overrides.Get("address").String())
+	assert.Equal(t, "0xe6330cf7", overrides.Get("functionSelector").String())
+	assert.Equal(
+		t,
+		"0x0000000000000000000000000000000000000000000000000000000000000002",
+		overrides.Get("dataPrefix").String())
+}
+
+func TestIntegration_FluxMonitor_NewRound(t *testing.T) {
+	app, cleanup := cltest.NewApplicationWithKey(t)
+	defer cleanup()
+
+	// 0. Start, connect, and initialize node
+	eth := app.MockCallerSubscriberClient(cltest.Strict)
+	eth.Context("app.StartAndConnect()", func(eth *cltest.EthMock) {
+		eth.Register("eth_getTransactionCount", `0x0100`) // TxManager.ActivateAccount()
+		eth.Register("eth_chainId", app.Store.Config.ChainID())
+	})
+	require.NoError(t, app.StartAndConnect())
+	eth.EventuallyAllCalled(t)
+
+	// 1. Configure fake Eth Node to return 10,000 cents when FM initiates price.
+	eth.Context("Flux Monitor initializes price", func(mock *cltest.EthMock) {
+		mock.Register("eth_call", "10000") // 10,000 cents
+		mock.Register("eth_call", "0x1")   // aggregator round: 1
+	})
+
+	// 2. Have price adapter server respond with 100 for price on initialization,
+	// NOT enough for deviation.
+	priceResponse := `{"data":{"result": 100}}`
+	mockServer, assertCalled := cltest.NewHTTPMockServer(t, http.StatusOK, "POST", priceResponse)
+	defer assertCalled()
+
+	// 3. Prepare new rounds logs subscription to be called by new FM job
+	newRounds := make(chan ethpkg.Log)
+	eth.RegisterSubscription("logs", newRounds)
+
+	// 4. Create FM Job, and ensure no runs because above criteria has no deviation.
+	buffer := cltest.MustReadFile(t, "testdata/flux_monitor_job.json")
+	var job models.JobSpec
+	err := json.Unmarshal(buffer, &job)
+	require.NoError(t, err)
+	job.Initiators[0].InitiatorParams.Feeds = models.Feeds([]string{mockServer.URL})
+
+	j := cltest.CreateJobSpecViaWeb(t, app, job)
+	_ = cltest.WaitForRuns(t, j, app.Store, 0)
+
+	eth.EventuallyAllCalled(t)
+
+	// 5. Send a NewRound log event to trigger a run.
+	log := cltest.LogFromFixture(t, "testdata/new_round_log.json")
+	attemptHash := cltest.NewHash()
+	confirmedReceipt := ethpkg.TxReceipt{
+		Hash:        attemptHash,
+		BlockNumber: cltest.Int(1),
+	}
+	eth.Context("ethTx.Perform() for new round send", func(eth *cltest.EthMock) {
+		eth.Register("eth_sendRawTransaction", attemptHash)         // Initial tx attempt sent
```

Comment:
Worth checking the `data` of the transaction?

Reply:
I believe that is already done in lower level tests:
1. flux_monitor_test.go: https://github.com/smartcontractkit/chainlink/blob/feature/169493405-fm-watches-rounds/core/services/flux_monitor_test.go#L386-L396
2. eth_tx_test.go: https://github.com/smartcontractkit/chainlink/blob/feature/169493405-fm-watches-rounds/core/adapters/eth_tx_test.go#L25

I'm not entirely sure how I would go about testing the specifics of the data here.  Let me know if you have a suggestion as to how to do this, if you think it's worth it. A link to another line of code works too.

Reply:
Was thinking using the callback feature of the mock to check args, but could also read from the DB. Seems ok with the lower level unit tests though.